framework/src/org/checkerframework/common/value/util/Range.java

```diff
@@ -0,0 +1,848 @@
+package org.checkerframework.common.value.util;
+
+import java.math.BigInteger;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import org.checkerframework.dataflow.util.HashCodeUtils;
+
+/**
+ * The Range class models a 64-bit two's-complement integral interval, such as all integers between
+ * 1 and 10, inclusive. Ranges are immutable.
+ *
+ * @author JasonMrX
+ */
+public class Range {
+
+    /** The lower bound of the interval, inclusive. */
+    public final long from;
+
+    /** The upper bound of the interval, inclusive. */
+    public final long to;
+
+    /** A range containing all possible 64-bit values. */
+    public static final Range EVERYTHING = new Range(Long.MIN_VALUE, Long.MAX_VALUE);
+
+    /** A range containing all possible 32-bit values values. */
+    public static final Range INT_EVERYTHING = new Range(Integer.MIN_VALUE, Integer.MAX_VALUE);
+
+    /** A range containing all possible 16-bit values values. */
+    public static final Range SHORT_EVERYTHING = new Range(Short.MIN_VALUE, Short.MAX_VALUE);
+
+    /** A range containing all possible 8-bit values values. */
+    public static final Range BYTE_EVERYTHING = new Range(Byte.MIN_VALUE, Byte.MAX_VALUE);
+
+    /** The empty range. */
+    public static final Range NOTHING = new Range(true);
+
+    /**
+     * Constructs a range with its bounds specified by two parameters, {@code from} and {@code to}.
+     *
+     * @param from the lower bound (inclusive)
+     * @param to the upper bound (inclusive)
+     */
+    public Range(long from, long to) {
+        if (!(from <= to)) {
+            throw new IllegalArgumentException(String.format("Invalid Range: %s %s", from, to));
+        }
+        this.from = from;
+        this.to = to;
+    }
+
+    /** Creates the singleton empty range. */
+    private Range(boolean isEmpty) {
```

Comment:
Why is the parameter necessary? There is no 0-ary overload and passing a boolean doesn't protect against abuse.
If it is useful, add a comment explaining why.